tests/server/signup.test.js

```diff
@@ -1,199 +1,156 @@
+      _callback(null, { lastID: 1 });
+    });
+
+    const response = await request.post(route)
+      .send({
+        username,
+        password
+      });
+
+    expect(response.statusCode).toBe(200);
   });
 
-  afterEach(async () => {
-    // delete id created in test
-    await db.run(`delete from User where user_id=${lastID + 1}`);
+  test("400 - Missing Username and Password", async () => {
+    const response = await request.post(route);
+
+    expect(response.statusCode).toBe(400);
+    expect(response.text).toBe(missingParametersText);
   });
-});
 
-const noUNOrPass = "Error: Username or password not set!";
+  test("400 - Missing Username", async () => {
+    const response = await request.post(route)
+      .send({
+        password
+      });
 
-describe("Test signup route", () => {
-  // block console logging
-  beforeAll(() => {
-    jest.spyOn(console, "log").mockImplementation(() => {});
-    jest.spyOn(console, "error").mockImplementation(() => {});
-  }); 
-  
-  test("Error 400 - no username or password", async () => {
-    const res = await request.post("/api/signup").send({});
-    expect(res.statusCode).toEqual(400);
-    expect(res.text).toEqual(noUNOrPass);
+    expect(response.statusCode).toBe(400);
+    expect(response.text).toBe(missingParametersText);
   });
 
-  test("Error 400 - no password", async () => {
-    const res = await request.post("/api/signup").send({
-      username: "username",
-    });
-    expect(res.statusCode).toEqual(400);
-    expect(res.text).toEqual(noUNOrPass);
+  test("400 - Missing Password", async () => {
+    const response = await request.post(route)
+      .send({
+        username
+      });
+
+    expect(response.statusCode).toBe(400);
+    expect(response.text).toBe(missingParametersText);
   });
 
-  test("Error 400 - no username", async () => {
-    const res = await request.post("/api/signup").send({
-      password: "password",
-    });
-    expect(res.statusCode).toEqual(400);
-    expect(res.text).toEqual(noUNOrPass);
+  test("400 - Invalid Username", async () => {
+    const response = await request.post(route)
+      .send({
+        username: "un$",
+        password: "Very1Epic!",
+      });
+
+    expect(response.statusCode).toBe(400);
   });
 
-  test("Error 400 - bad username", async () => {
-    const res = await request.post("/api/signup").send({
-      username: "un$",
-      password: "Very1Epic!",
-    });
-    expect(res.statusCode).toEqual(400);
+  test("400 - Invalid Password", async () => {
+    const response = await request.post(route)
+      .send({
+        username: "usernameVeryEpic",
+        password: "invalid"
+      });
+
+    expect(response.statusCode).toBe(400);
   });
 
-  test("Error 400 - bad password", async () => {
-    const res = await request.post("/api/signup").send({
-      username: "usernameVeryEpic",
-      password: "password",
+  test("400 - Existing Username", async () => {
+    db.run = jest.fn((query, params, _callback) => {
```

Comment:
Confused why you're using `_callback` when in every other instance it is `callback`.

Reply:
Same at line 128-129.

Reply:
probably copy-pasted and forgot to change